src/libtoast/src/toast_math_linearalgebra.cpp

```diff
@@ -32,18 +32,20 @@ void toast::LinearAlgebra::gemm(char TRANSA, char TRANSB, int M, int N,
                                 int LDC) {
 #ifdef HAVE_CUDALIBS
 
+    auto & pool = GPU_memory_pool::get();
+
     // prepare inputs
     cublasOperation_t transA_gpu = (TRANSA == 'T') ? CUBLAS_OP_T : CUBLAS_OP_N;
     cublasOperation_t transB_gpu = (TRANSB == 'T') ? CUBLAS_OP_T : CUBLAS_OP_N;
 
     // send data to GPU
-    double * A_gpu = GPU_memory_pool.toDevice(A, LDA * ((TRANSA == 'N') ? K : M));
-    double * B_gpu = GPU_memory_pool.toDevice(B, LDB * ((TRANSB == 'N') ? N : K));
-    double * C_gpu = (BETA == 0.) ? GPU_memory_pool.alloc <double> (LDC * N)
-                                 : GPU_memory_pool.toDevice(C, LDC * N);
+    double * A_gpu = pool.toDevice(A, LDA * ((TRANSA == 'N') ? K : M));
```

Comment:
I'm not a fan of this syntax. It is concise but usually makes the code less readable. Obviously a matter of preference.

Reply:
Can you clarify a bit?  Are you talking about the original function signature (which is unmodified in this PR) or the use of a singleton class rather than a global variable?

Reply:
The original signature with the in-line conditional.